app/src/main/java/org/jellyfin/androidtv/details/EpisodeDetailsFragment.kt

```diff
@@ -0,0 +1,78 @@
+package org.jellyfin.androidtv.details
+
+import androidx.leanback.widget.ArrayObjectAdapter
+import androidx.leanback.widget.ClassPresenterSelector
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.withContext
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.details.actions.*
+import org.jellyfin.androidtv.model.itemtypes.Episode
+import org.jellyfin.androidtv.presentation.InfoCardPresenter
+import org.jellyfin.androidtv.util.ImageUtils
+import org.jellyfin.androidtv.util.addIfNotEmpty
+import org.jellyfin.androidtv.util.apiclient.getEpisodesOfSeason
+import org.jellyfin.androidtv.util.dp
+import org.jellyfin.apiclient.model.entities.PersonType
+
+class EpisodeDetailsFragment(private val episode: Episode) : BaseDetailsFragment<Episode>(episode) {
+	// Action definitions
+	private val actions by lazy {
+		val item = MutableLiveData(episode)
+
+		listOf(
+			ResumeAction(context!!, item),
+			PlayFromBeginningAction(context!!, item),
+			ToggleWatchedAction(context!!, item),
+			ToggleFavoriteAction(context!!, item),
+
+			// "More" button
+			SecondariesPopupAction(context!!, listOfNotNull(
+				episode.seasonId?.let {GoToItemAction(context!!, context!!.getString(R.string.lbl_goto_season), it) },
+				episode.seriesId?.let {GoToItemAction(context!!, context!!.getString(R.string.lbl_goto_series), it) },
+				DeleteAction(context!!, item) { activity?.finish() }
+			))
+		)
+	}
+
+	// Row definitions
+	private val detailRow by lazy {
+		val primary = episode.images.parentPrimary ?: episode.seasonPrimaryImage ?: episode.seriesPrimaryImage ?: episode.images.primary
+		val backdrops = episode.images.parentBackdrops ?: episode.images.backdrops
+		DetailsOverviewRow(episode, actions, primary, backdrops)
+	}
+	private val moreFromThisSeason by lazy { createListRow("More from this Season", emptyList(), ItemPresenter(context!!, (ImageUtils.ASPECT_RATIO_16_9 * 140.dp).toInt(), 140.dp, true)) }
+	private val guestStars by lazy { createListRow("Guest Stars", episode.cast.filter { it.type == PersonType.GuestStar }, PersonPresenter(context!!)) }
```

Comment:
Hardcoded string

Reply:
Btw, why not the full cast?

Reply:
That was the behaviour of the old screen, I don't have a preference on it

Reply:
Let's just show the full cast

Reply:
Hm, it seems like we're not sent a cast, but I now implemented it in a way that guest stars are in one row if there are any, and remaining cast in a second one

Reply:
Turns out we would be if it were implemented. Changed it to have two rows, one for Guest Stars and one for the normal cast without them.